app/controllers/oauth_authorized_applications_controller.rb

```diff
@@ -0,0 +1,14 @@
+class OauthAuthorizedApplicationsController < Doorkeeper::AuthorizedApplicationsController
+
+  helper_method :dashboard_or_root_path
+
+  protected
+
+  def dashboard_or_root_path
```

Comment:
yes, this was totally a hack just to get it working. there's duplicate code in these three files. I am still unclear what the solution is after your comment, but I'll have a go at it soon.